### PR TITLE
Add circleci config

### PR DESCRIPTION
**Pull request by SevereOverfl0w**
See passing [here](https://app.circleci.com/pipelines/github/SevereOverfl0w/test.contract/2/workflows/bace9ad4-8eb8-42fd-8012-1fb6522c1387/jobs/1).  Will need further setup to get enabled for the repo.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  build:
+    working_directory: ~/repo
+    docker:
+      - image: cimg/clojure:1.11
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - deps-{{ checksum "deps.edn" }}
+            - deps-
```
Comment (miridius) [inline .circleci/config.yml:12]: Does this line do anything?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - run:
+          name: Download Dependencies
+          command: clojure -R:test -Spath
```
Comment (miridius) [inline .circleci/config.yml:15]: Issue?: this doesn't work for me locally:

```
❯ clojure -R:test -Spath      
-R is no longer supported, use -A with repl, -M for main, -X for exec, -T for tool
```

Maybe we want `clojure -Atest -Spath`?

**Comment by arohner** [comment]
The test command should be `bazel test //...`, which then means the image will need bazel on it. I'll approve this to unblock it, but please convert it to bazel

**Review by arohner — APPROVED** [review]

**Review by miridius — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (SevereOverfl0w) [reply .circleci/config.yml:15]: I don't think they actually have a supported fallback for this yet.

File: .circleci/config.yml (hunk 1 above)
Comment (SevereOverfl0w) [reply .circleci/config.yml:12]: It's a fallback cache to use when the other one is invalid.  It'll grab whatever the newest deps-* is.

Comment (miridius) [reply .circleci/config.yml:12]: oh nice!

**Pull request merged**